gamemodes/terrortown/entities/weapons/weapon_tttbase.lua

```diff
@@ -128,6 +128,10 @@ SWEP.silentPickup = false
 -- Can be useful if you have multiple instances, that rely on global variables stored via weapons.GetStored()
 SWEP.HotReloadableKeys = {}
 
+-- Auto select the last weapon, when this weapon is removed from the player's iventory
+-- Only works if OnRemove in the base class is called
+SWEP.SelectLastWeaponOnRemove = true
```

Comment:
I dont think this needs to be adjustable. If somebody doesnt want it, just override OnRemove.

Reply:
I'm a fan of explicit variables to explain what happens here, but if I'm alone with that, I can remove that

Reply:
Not sure what you mean with explicit variables. But this is not really an attribute a weapon needs imho. Normally the selectlastweapononremove shouldnt even be handled by a weapon itself, but by the player-entity.
Also using a consolecommand seems like a lazy old global call to something you dont want to get the reference for.

But this is just me seeing those few lines. I didnt investigate the overall workings closely, just what I expect it to do.

Reply:
> Normally the selectlastweapononremove shouldnt even be handled by a weapon itself, but by the player-entity.

But it has to trigger on weapon remove, that's easily done in the weapon base. I could also put this in a global hook, that would probably work as well.

> Also using a consolecommand seems like a lazy

I think that is honestly the best way to do. Otherwise I would have to track the weapon switches so I always know the last valid weapon. There is no function to get the last valid weapon.
Example: If I switch from A to B to C and B is removed before C, then I should auto switch back to A. So the tracking would be a bit more complex. GMod already does that, but only exposes this as a clientside command.

Another thing is that for some reason you can't trigger the selection of a weapon in the same tick you give it to a player. Even a tick later is not late enough. My testing showed that you need at least a delay of 0.2 seconds. Which is kinda weird.
This would probably not be an issue here, but something that you have to keep in mind

Reply:
Yeah i get that, I sometimes forget, that we have to deal with gmod in many of those weird cases.

Still think the same about removing that variable.

Reply:
I can remove it, it is fine by me

Reply:
done